### ISSUE TITLE
Old (uninstalled) Theme Throwing Errors

### ISSUE THREAD

**Issue by bobchristenson** [backdrop/backdrop-issues#5073]
I migrated a site from D7 -> Backdrop and previously had the Zurb Foundation theme installed in D7.  When I moved to Backdrop I removed that theme.  I don't remember, honestly, if I uninstalled it in Drupal or whether I uninstalled it in Backdrop.  Regardless, it was removed and the files no longer exist in the themes directory at all.

Yet, I'm getting an error constantly in the logs:
`Notice: Undefined index: zurb_foundation in theme_has_settings() (line 1490 of /home/core/includes/theme.inc).`

I've tried the normal things (clearing caches, visiting/saving the themes page, etc).  I still get the error.  Where is the call to zurb_foundation coming from if I don' t have the theme installed anymore?

**Comment by bobchristenson** [comment]
Weird. I had an old D7 theme in my themes directory that was using foundation...but it wasn't enabled and wasn't even ported to Backdrop. I removed that and the errors went away. Not sure whats up with that...

Will continue to monitor this to see if the errors pop back up.

**Comment by indigoxela** [comment]
@bobchristenson many thanks for reporting.

> I had an old D7 theme in my themes directory that was using foundation...but it wasn't enabled

Strange indeed. Do you have direct access to the database? Like, for instance, with phpmyadmin?

Try: `SELECT * FROM system WHERE type = 'theme';` Does it still show up? As enabled (in the "status" column)?

If so, you might have forgotten to disable it in D7. If not, the problem should be solved.

**Comment by bobchristenson** [comment]
Nope, the Foundation theme wasn't listed in my system table.   Not sure if that might get automatically removed after I remove the theme files (?) but I don't see it there now.

**Comment by indigoxela** [comment]
> ...but it wasn't enabled and wasn't even ported to Backdrop

I'm not sure, but maybe we should make the theme recognition a bit more robust, or should I say picky in what's considered a valid Backdrop theme?

**Comment by bobchristenson** [comment]
Yeah, I would think it should have a valid Backdrop info file to even be looked at (?).  This isn't very helpful, but it's POSSIBLE i had the Backdrop version of Foundation installed when I first tried to get this running. It's possible that this scenario occurred:

1. I migrated the site, Drupal version of Foundation in place.  Backdrop didn't detect it.
2. I removed the Drupal version, put the Backdrop version of Foundation in place and enabled it.
3. Removed the Backdrop version (because it didn't work great) and put the Drupal version back.

At that point, it may have already recognized the Backdrop version of Foundation and thought that the Drupal version was still that Backdrop version (if it had the same machine name for the theme?).  I don't know.  It's not helpful because I don't know exactly what order I did all that...sorry I can't offer more clarity, but the above guess seems to make a bit more sense.

**Comment by indigoxela** [comment]
Ha! We already tried to fix this: https://github.com/backdrop/backdrop-issues/issues/3981

I wonder if this fix would also prevent the problem you describe... I'll have a closer look. However, I'm closing this one as "duplicate" - I hope that's OK for you @bobchristenson

**Comment by klonos** [comment]
@indigoxela I would still like to try to reproduce this, and make sure that it is not due to human error (forgetting to uninstall the theme in D7), plus I don't think it's exactly the same as #3981.

The error thrown here is around `theme_has_settings()`, whereas the other issue is around `theme_get_setting()`. They could be related, but perhaps not.
